### DISCUSSION TITLE
Descrete volume and weights for items

### DISCUSSION THREAD

**Discussion by olanti-p** [cataclysmbn/Cataclysm-BN#2505] [General]
Stuff like this
```
    "id": "blanket",
    "type": "ARMOR",
    "name": { "str": "blanket" },
    "description": "Hiding under here will not protect you from the monsters.",
    "weight": "1123 g",
```
Is all over the place.

Many items use _suspiciously precise_ numbers for their weights and/or volumes that don't convert nicely into either metric or the other measurement system.

Not only the UI cannot display such precision:
![image](https://user-images.githubusercontent.com/60584843/228894748-5bbf2007-4cd8-4825-ba5f-471668f2b1b8.png) ![image](https://user-images.githubusercontent.com/60584843/228895226-bcefeca6-a6a2-4a83-8fd8-e432271eaa52.png)

But since we're not pursuing realism, there's also literally zero gameplay value in having a `2002 grams` item over a `2000 grams` one. On the contrary, round numbers are much easier to calculate in mind if for example the player would want to estimate how much 3 or 4 copies of an item would weight.

Thus, the proposal is to:
1. Choose a discrete value for volumes and weights.
    `10 grams` / `10 milliliters` may be a good starting point, as it the smallest the UI can display.
    Older Cata used `250 milliliters`, not sure whether we want to go this far.
2. Update weight and volume in all items with a script
3. Ammo items may need special handling, as their volume is defined per stack but their weight - per unit
4. (Maybe) Add a check for discrete values on item load

**Comment by chaosvolt** [comment]
Some of these values are for the sake of showing as exactly X pounds, since that's the default weight unit used. Doesn't apply to oddball volume units however, since those are in liters by default.

**Reply by scarf005** [reply]
<img src="https://github.com/user-attachments/assets/6f773ff1-4365-446d-a0ab-3c76c3570af7" width="50%">

since #2510 it's the other way around tho, now kg is the default

**Comment by scarf005** [comment]
was writing script to find and replace such values to something more sane, e.g 1kg/1250g. also about pounds, should we change it to 450g or 500g?

**Reply by chaosvolt** [reply]
540 is closer, since that's about 453.592 grams. Round-to-nearest is probably best approach for stuff like that.
